pkg/updater/resultstore/resultstore.go

```diff
@@ -453,6 +457,75 @@ func processGroup(tg *configpb.TestGroup, group *invocationGroup) *updater.Infla
 	return col
 }
 
+// getTestResults traverses through a test suite and returns a list of all tests
+// that exists inside of it.
+func getTestResults(testSuite *resultstorepb.TestSuite) []*resultstorepb.Test {
+	if testSuite == nil || testSuite.Tests == nil {
+		return nil
+	}
+
+	if len(testSuite.GetTests()) == 0 {
+		return []*resultstorepb.Test{
+			{
+				TestType: &resultstorepb.Test_TestSuite{TestSuite: testSuite},
+			},
+		}
+	}
+
+	var tests []*resultstorepb.Test
+	for _, test := range testSuite.GetTests() {
+		if test.GetTestCase() != nil {
+			tests = append(tests, test)
+		} else {
+			tests = append(tests, getTestResults(test.GetTestSuite())...)
+		}
+	}
+	return tests
+}
+
+func testMethodLimit(tg *configpb.TestGroup) int {
+	var testMethodLimit int
+	const defaultTestMethodLimit = 20
+	if tg == nil {
+		return defaultTestMethodLimit
+	}
+	if tg.EnableTestMethods {
+		testMethodLimit = int(tg.MaxTestMethodsPerTest)
+		if testMethodLimit == 0 {
+			testMethodLimit = defaultTestMethodLimit
+		}
+	}
+	return testMethodLimit
+}
+
+// processTestResults iterates through a list of test results and adds them to
+// a map of groupedcells based on the method name produced
+func processTestResults(tg *config.TestGroup, testResults []*resultstorepb.Test, cell updater.Cell, targetID string, sar *singleActionResult, testMethodLimit int, groupedCells map[string][]updater.Cell) {
+	for _, testResult := range testResults {
+		var methodName string
+		if testResult.GetTestCase() != nil {
+			methodName = testResult.GetTestCase().GetCaseName()
+		} else {
+			methodName = testResult.GetTestSuite().GetSuiteName()
+		}
+
+		if tg.UseFullMethodNames {
+			parts := strings.Split(testResult.GetTestCase().GetClassName(), ".")
+			className := parts[len(parts)-1]
+			methodName = className + "." + methodName
+		}
+
+		methodName = targetID + "@TESTGRID@" + methodName
+
+		if cell.Result == statuspb.TestStatus_PASS_WITH_SKIPS && tg.IgnoreSkip {
+			continue
+		}
+		if len(testResults) <= testMethodLimit {
+			groupedCells[methodName] = append(groupedCells[methodName], cell)
+		}
```

Comment:
you should create a separate testResultCell and append it to the groupedCells. you can create one like this:
```
trCell := Cell{
  ID: targetID, // same targetID as the parent TargetResult
  CellID: cell.CellID, // same cellID
  Result: testResultStatus(testResult) // you have to perform mapping of statuses
}
```